### PR TITLE
Refactor intra doc link code

### PR DESCRIPTION
**Pull request by jyn514**
I got tired of `fold_item` being 500 lines long.
This is best reviewed one commit at a time with whitespace changes hidden.
There are no logic changes other than the last commit making a parameter checked by the caller instead of the callee.

r? @Manishearth

### CODE REVIEW

**Comment by jyn514** [comment]
The number of parameters is kind of awful ... I might add a struct that groups them into one.

**Comment by jyn514** [comment]
> The number of parameters is kind of awful ... I might add a struct that groups them into one.

Yeah no this had way too many compile errors for me to deal with right now, if I do add it will be in a follow-up Pr.

File: src/librustdoc/passes/collect_intra_doc_links.rs [hunk 1]
```diff
@@ -998,169 +998,19 @@ impl LinkCollector<'_, '_> {
...
-                                disambiguator,
-                                dox,
-                                link_range,
-                                smallvec![kind],
-                            );
-                            return;
-                        }
-                    }
-                }
+            match self.resolve_with_disambiguator(
```
Comment (Manishearth) [inline src/librustdoc/passes/collect_intra_doc_links.rs:989]: This is just `?`

**Review by Manishearth — APPROVED** [review]
lgtm

File: src/librustdoc/passes/collect_intra_doc_links.rs (hunk 1 above)
Comment (jyn514) [reply src/librustdoc/passes/collect_intra_doc_links.rs:989]: This returns `()`, not `Option`. So it can't use `?`.

**Comment by jyn514** [comment]
@bors r=manishearth

Note to anyone making rollups: this is unlikely to fail any tests, but is extremely likely to conflict with other intra-doc link PRs.

**Pull request merged**